src/shared/helpers/api_helper.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Helper functions shared between components
+ */
+
+// Enum for HTTP methods
+export enum Method {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    DELETE = 'DELETE',
+}
+
+export const request = async (url: string, method: Method, body?: any): Promise<any | undefined> => {
+    try {
+        const response = await fetch(url, {
```

Comment:
FYI: It's largely abstracted away from every day use, but we use a library called `@tanstack/react-query` for our querying: https://tanstack.com/query/v4/docs/framework/react/overview

It might be worth having a look and seeing if you can refactor one or two of your queries to use it, just so you know how to.

We also tend to abstract these queries into hooks, similar to how you've abstracted them into services, I can try and find you an example in the codebase if you can't find one.